swingset/src/main/java/com/nqadmin/swingset/SSDBComboBox.java

```diff
@@ -1306,6 +1269,12 @@ public void setSecondDisplayColumnName(final String _secondDisplayColumnName) {
 	 *
 	 * @param _value value to set as currently selected.
 	 */
+	//
+	// TODO: The code currently assumes that _value is an SSListItem.
+	//       But the description above makes me dubious of that claim.
+	//       Is _value a string? Should the type be checked and
+	//       and then proceed accordingly?
+	//
 	@Override
 	public void setSelectedItem(final Object _value) {
```

Comment:
There may be some work needed around here.

Reply:
Prior to SwingSet 4, setSelectedItem() was only ever called once and that was with null. I will see if/how it is used in our app. It was not overridden previously and I think it was only overridden here to work around odd behvaior with GlazedLists. Looks like settingSelectedItem() has been commented out so I'm thinking we may not even need to override setSelectedItem() any more??

Reply:
No, maybe it's sitll needed. I didn't scroll down enough. The selectAll() is useful, although it's possible we could relocate it.

Reply:
For some random reason I put in
```
SSListItem newSelectedItem = (SSListItem) _value;
```
Changing this to
```
Object newSelectedItem = _value;
```
And I think everything works ok. Escpecially if don't want to go into `== null` branch.

Reply:
It's not used in our application code at all.

Reply:
Not used in our app, although I guess someone might use it since we're overriding a public method so it should probably behave.